ext/node/polyfills/worker_threads.ts

```diff
@@ -348,6 +350,10 @@ internals.__initWorkerThreads = (
       const { 0: metadata, 1: _ } = maybeWorkerMetadata;
       workerData = metadata.workerData;
       environmentData = metadata.environmentData;
+      const env = metadata.env;
+      if (env) {
+        process.env = env;
+      }
```

Comment:
This is bit strange - does it mean that setting a new property won't have an effect on the whole process? Can we add a test that checks this behavior?